app/models/location.rb

```diff
@@ -402,17 +407,38 @@ def reactivating_soon?
     status == :reactivating_soon
   end
 
-private
-
-  PIO = PostcodeService.new
-
   def validate_postcode
-    if postcode.nil? || !postcode&.match(POSTCODE_REGEXP)
+    if postcode.blank?
+      error_message = I18n.t("validations.location.postcode_blank")
+      errors.add :postcode, error_message
+    elsif !postcode&.match(POSTCODE_REGEXP)
       error_message = I18n.t("validations.postcode")
       errors.add :postcode, error_message
     end
   end
 
+  def validate_location_admin_district
+    if location_admin_district == "Select an option"
+      error_message = I18n.t("validations.location_admin_district")
+      errors.add :location_admin_district, error_message
+    end
+  end
```

Comment:
Remove in favour of standard `presence: true` validation once you're rejecting the select option value at the params level.